File: Debt.cs
using System;

class Icon
{
    float speed, glow, energy;
    int x, y;
    int subtype; // spinner, slider or hopper

    bool clockwise; // need for spinner
    bool expand;    // need for spinner
    bool vertical;  // need for slider

    int distance;   // need for slider
    bool visible;   // need for hopper

    int xcoord, ycoord; // need for hopper

    void Spin() { }

    void Slide() { }

    void Hop() { }

    // constructor must set subtype: client must pass value
    public Icon(uint value)
    {
        subtype = (int)value; // use enum for readability
        // and then use conditional to set associated fields
    }

    public void Move()
    {
        if (subtype == 1)
        {
            Spin();
        }
        else if (subtype == 2)
        {
            Slide();
        }
        else
        {
            Hop();
        }
    }

    // tedious subtype checking: subtype drives flair details
    public void Flair()
    {
        if (subtype == 1)
        {
            Spin();
        }
        else if (subtype == 2)
        {
            Slide();
        }
        else
        {
            Hop();
        }
    }
}

File: Debt.py
class Icon:
    def __init__(self, value: int):
        self.speed = 0.0
        self.glow = 0.0
        self.energy = 0.0
        self.x = 0
        self.y = 0
        self.subtype = value  # spinner, slider or hopper

        # spinner
        self.clockwise = False
        self.expand = False

        # slider
        self.vertical = False
        self.distance = 0

        # hopper
        self.visible = False
        self.xcoord = 0
        self.ycoord = 0

        # constructor must set subtype: client must pass value
        # use enum for readability in real design

    def spin(self):
        pass

    def slide(self):
        pass

    def hop(self):
        pass

    def move(self):
        if self.subtype == 1:
            self.spin()
        elif self.subtype == 2:
            self.slide()
        else:
            self.hop()

    def flair(self):
        if self.subtype == 1:
            self.spin()
        elif self.subtype == 2:
            self.slide()
        else:
            self.hop()
